modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -50,13 +50,17 @@ export class {{classname}} {
      * {{notes}}
      {{#allParams}}* @param {{paramName}} {{description}}
      {{/allParams}}*/
-    public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}extraHttpRequestParams?: any): Observable<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
+    public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}extraHttpRequestParams?: any): Observable<undefined | {{#responses}}{{#dataType}}{{{dataType}}} | {{/dataType}}{{/responses}}{}> {
```

Comment:
@damienpontifex seems to me the method signature has changed and it should be filed against 2.3.0 instead.

Also there's another PR (merged) that updated the method signature in a similar way: https://github.com/swagger-api/swagger-codegen/pull/5972. Please pull the latest 2.3.0 to take a look.